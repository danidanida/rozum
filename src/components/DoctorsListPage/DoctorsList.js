import React from "react";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import { Link } from "react-router-dom";

const DoctorsListTable = (props) => {
  const doctors = props.doctors;
  const path = "/doctors/";

  return (
    <div>
      <TableContainer component={Paper}>
        <Table aria-label="simple table">
          <TableHead>
            <TableRow>
              <TableCell align="center">ФИО</TableCell>
              <TableCell align="center">Дата рождения</TableCell>
              <TableCell align="center">Телефон</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {doctors.map((row, i) => (
              <TableRow key={i}>
                <TableCell align="center">
                  <Link to={path + row.id}>
                    {`${row.lastName} ${row.firstName} ${row.middleName}`}
                  </Link>
                </TableCell>
                <TableCell align="center">{row.birthDate}</TableCell>
                <TableCell align="center">{row.phone}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </div>
  );
};

export default DoctorsListTable;
