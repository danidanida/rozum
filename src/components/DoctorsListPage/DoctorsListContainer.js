import React from "react";
import { connect } from "react-redux";
import DoctorListTable from "./DoctorsList";
import { compare, reformatDate } from "./utils";

const DoctorsList = (props) => {
  if (!props.initialized) {
    return <h2>Loading...</h2>;
  }

  const table = <DoctorListTable doctors={props.doctorsList} />;

  return <div> {table} </div>;
};

const mapStateToProps = (state) => {
  let sortedDoctorsArray = [];

  sortedDoctorsArray = state.doctorsList.sort(compare).map((d) => {
    return { ...d, birthDate: reformatDate(d.birthDate) };
  });

  return { doctorsList: sortedDoctorsArray, initialized: state.initialized };
};

export default connect(mapStateToProps)(DoctorsList);
