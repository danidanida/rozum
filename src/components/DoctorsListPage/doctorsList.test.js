import React from "react";
import { MemoryRouter } from "react-router-dom";
import { mapStateToProps } from "./DoctorsListContainer";
import renderer from "react-test-renderer";
import DoctorsList from "./DoctorsList";

describe(`Doctors List Page`, () => {
  const appState = {
    doctorsList: [
      {
        id: 1,
        firstName: "Michel",
        lastName: "B",
        middleName: "M",
        birthDate: "1678-05-27",
        phone: "+79975669",
      },
      {
        id: 2,
        firstName: "Sam",
        lastName: "S",
        middleName: "K",
        birthDate: "2020-09-18",
        phone: "+7495",
      },
    ],
  };

  describe("MapStateToProps", () => {
    it("Should map the state to props correctly", () => {
      const ownProps = {};
      const componentState = mapStateToProps(appState, ownProps);
      expect(componentState.doctorsList[0].birthDate).toBe("27/05/1678");
    });
  });

  describe("Doctors list", () => {
    it("should not regress", () => {
      const tree = renderer.create(
        <MemoryRouter>
          <DoctorsList doctors={appState.doctorsList} />
        </MemoryRouter>
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
