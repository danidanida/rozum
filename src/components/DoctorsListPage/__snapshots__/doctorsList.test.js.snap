// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doctors List Page Doctors list should not regress 1`] = `
<div>
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            ID
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            ФИО
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            Дата рождения
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
          >
            1
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
          >
            <a
              href="/doctors/1"
              onClick={[Function]}
            >
              B Michel M
            </a>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
          >
            1678-05-27
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
          >
            2
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
          >
            <a
              href="/doctors/2"
              onClick={[Function]}
            >
              S Sam K
            </a>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
          >
            2020-09-18
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
